packages/block-library/src/buttons/edit.js

```diff
@@ -1,7 +1,10 @@
 /**
  * WordPress dependencies
  */
-import { InnerBlocks } from '@wordpress/block-editor';
+import {
+	__experimentalAlignmentHookSettingsProvider as AlignmentHookSettingsProvider,
```

Comment:
I guess my thinking was that it's the responsibility of the "buttons" to create the context and provide it while the "button" would consume it. And this is possible with the private API.

It seems the fact that we use the "align support" make this impossible.

Reply:
@youknowriad Maybe we should implement a hook for alignment, similar to the `useColors` hook, and start using that everywhere instead of the `supports` flag. Would that make what you want to do possible?